daq/faucet_state_collector.py

```diff
@@ -308,21 +370,12 @@ def get_access_switch(self, mac):
         learned_switches = self.learned_macs.get(mac, {}).get(KEY_MAC_LEARNING_SWITCH)
 
         for switch, port_map in learned_switches.items():
-            access_switch_port[switch] = port_map[KEY_MAC_LEARNING_PORT]
-
+            port = port_map[KEY_MAC_LEARNING_PORT]
+            port_attr = self.get_port_attributes(switch, port)
+            if port_attr['type'] == 'access':
+                access_switch_port[switch] = port
         if not access_switch_port:
-            return None
-
-        for link_map in self.topo_state.get(TOPOLOGY_GRAPH).get("links", []):
-            if not link_map:
-                continue
-
-            sw_1, port_1, sw_2, port_2 = FaucetStateCollector.get_endpoints_from_link(link_map)
-            if access_switch_port.get(sw_1, "") == port_1:
-                access_switch_port.pop(sw_1)
-            if access_switch_port.get(sw_2, "") == port_2:
-                access_switch_port.pop(sw_2)
-
+            return None, None
```

Comment:
Can we do it like this? If found, return the switch and port in the for loop. Otherwise, return here without checking "if not access_switch_port"